opentelemetry-sdk/src/opentelemetry/sdk/_metrics/aggregation.py

```diff
@@ -14,124 +14,198 @@
+            value = self._value
+            start_time_unix_nano = self._start_time_unix_nano
 
-class SumAggregation(Aggregation):
-    """
-    This aggregation collects data for the SDK sum metric point.
-    """
+            self._value = 0
+            self._start_time_unix_nano = now + 1
 
-    def __init__(self, instrument):
-        self._value = 0
+        return Sum(
+            aggregation_temporality=AggregationTemporality.DELTA,
+            is_monotonic=self._is_monotonic,
+            start_time_unix_nano=start_time_unix_nano,
+            time_unix_nano=now,
+            value=value,
+        )
 
-    def aggregate(self, value):
-        self._value = self._value + value
 
-    def make_point_and_reset(self):
-        pass
+class AsynchronousSumAggregation(
+    _MonotonicityAwareAggregation, Aggregation[Sum]
+):
+    def __init__(self, is_monotonic: bool):
+        super().__init__(is_monotonic)
+        self._value = None
+        self._start_time_unix_nano = _time_ns()
 
+    def aggregate(self, measurement: Measurement) -> None:
+        with self._lock:
+            self._value = measurement.value
 
-class LastValueAggregation(Aggregation):
+    def collect(self) -> Optional[Sum]:
+        """
+        Atomically return a point for the current value of the metric.
+        """
+        if self._value is None:
+            return None
+
+        return Sum(
+            start_time_unix_nano=self._start_time_unix_nano,
+            time_unix_nano=_time_ns(),
+            value=self._value,
+            aggregation_temporality=AggregationTemporality.CUMULATIVE,
+            is_monotonic=self._is_monotonic,
+        )
 
-    """
-    This aggregation collects data for the SDK sum metric point.
-    """
 
-    def __init__(self, instrument):
+class LastValueAggregation(Aggregation[Gauge]):
+    def __init__(self):
+        super().__init__()
         self._value = None
-        self._timestamp = _time_ns()
 
-    def aggregate(self, value):
-        self._value = value
-        self._timestamp = _time_ns()
-
-    def make_point_and_reset(self):
-        pass
+    def aggregate(self, measurement: Measurement):
+        with self._lock:
+            self._value = measurement.value
 
+    def collect(self) -> Optional[Gauge]:
+        """
+        Atomically return a point for the current value of the metric.
+        """
+        if self._value is None:
+            return None
 
-class ExplicitBucketHistogramAggregation(Aggregation):
+        return Gauge(
+            time_unix_nano=_time_ns(),
+            value=self._value,
+        )
 
-    """
-    This aggregation collects data for the SDK sum metric point.
-    """
 
+class ExplicitBucketHistogramAggregation(
+    _MonotonicityAwareAggregation, Aggregation[Histogram]
+):
     def __init__(
         self,
-        instrument,
-        *args,
-        boundaries=(0, 5, 10, 25, 50, 75, 100, 250, 500, 1000),
-        record_min_max=True,
+        is_monotonic: bool,
```

Comment:
Imo it is really confusing to say a histogram `is_monotonic`, since monotonicity pertains to a sequence of numbers. In this case I get that the `is_monotonic` means the `sum` field is monotonic, but it reads as if the actual sequence of measurements this aggregator sees is monotonic.

For example, if you were recording temperatures and you saw `-10, -9, -8`, that is a monotonic sequence of temperatures, even though the sum is decreasing.

It might make more sense to have a bool flag `received_negative` and set it true if we actually see negatives (we might do this for `SumAggregation` too?)

Reply:
The concept of monotonicity does not apply to aggregations, but to their corresponding instruments. The current attribute name is misleading and can easily cause confusion, I have renamed the attribute to be more clear.

Thanks for pointing that out! :v:

Reply:
I think it applies to the aggregation, but not to histogram. Histograms should only be positive for now anyway. I saw we handle that error when validating views and remove any monotonicity related stuff from ExplicitBucketHistogram

Reply:
From 1/11 meeting, will be removing monotonic out of this class.